### PR TITLE
RTS-1051: Clean up a few outstanding issues

### PR DESCRIPTION
**Pull request by hazen**
Both @hmmr and @andytill found a couple of optimizations, which have been addressed here.

### CODE REVIEW

**Comment by hmmr** [comment]
+1 e31b887

File: src/riak_kv_compile_tab.erl [hunk 1]
```diff
@@ -169,13 +169,9 @@ get_ddl(BucketType) when is_binary(BucketType) ->
 %%
 -spec get_all_table_names() -> term() | notfound.
```
Comment (andytill) [inline src/riak_kv_compile_tab.erl]: The spec here is incorrect.

**Comment by hmmr** [comment]
+1 1ad180f

**Comment by hazen** [comment]
@borshop merge

**Pull request merged**